### ISSUE TITLE
(3) MS 20: Fix Image file and thumbnail mismatch

### ISSUE THREAD

**Issue by chesleyperry** [UCSCLibrary/dams_project_mgmt#216]
### Descriptive Summary
Originally we had thought this was a metadata/image file mismatch resulting from the mass digitization workflow, but it turns out that this is not the case.  Rather, there is a mismatch between [the thumbnail that is being displayed on the search results page]( https://digitalcollections.library.ucsc.edu/catalog?utf8=%E2%9C%93&locale=en&search_field=all_fields&q=ms0020_neg_4764.tif) (which shows the correct image or PJ with his camera case) and the image that appears on the [item landing page ](http://digitalcollections.library.ucsc.edu/concern/parent/6969z3268/works/r781wg644)(which is not the correct image; rather, it's an image of an unrelated drawing).

In checking the image file (ms0020_neg_4764.tif) and metadata on dams_ingest, we have confirmed that 1) ms0020_neg_4764.tif shows PJ sitting with his camera case, and 2) the image matches the metadata it was ingested with.

### Acceptance Criteria
- [x] Image appearing on [the item page](http://digitalcollections.library.ucsc.edu/concern/parent/6969z3268/works/r781wg644) is ms0020_neg_4764.tif

### Background
**Original incorrect description of this issue:**
The work at http://digitalcollections.library.ucsc.edu/concern/parent/6969z3268/works/r781wg644
Filename ms0020_neg_4764.tif 
has the wrong title.  The image matches the images in the subseries "[Untitled: Copy series for 1947 C.S.F.A.assignment](http://digitalcollections.library.ucsc.edu/catalog?locale=en&q=Copy+series+for+1947+C.S.F.A.assignment+%231&search_field=subseries)"

Metadata for the work should be updated - not a huge priority.

**Comment by rmjaffe** [comment]
@Sue I went back to the re-ingest spreadsheet and back to the original metadata spreadsheet and this file (ms0020_neg_4764.tif) is associated with this title (Pirkle Jones sitting with camera case, San Francisco) on both.

@katedundon @tmariamora Can you confirm that this image (ms0020_neg_4764.tif) matches [Untitled: Copy series for 1947 C.S.F.A.assignment #1](Untitled: Copy series for 1947 C.S.F.A.assignment #1) and [Untitled: Copy series for 1947 C.S.F.A.assignment #2](http://digitalcollections.library.ucsc.edu/records/6969z139x)?  And if so, what should the title for ms0020_neg_4764.tif be, what should the rest of the metadata record look like?

**Comment by rmjaffe** [comment]
@tmariamora @katedundon Gentle reminder:  When you have a moment could you look into this?

**Comment by tmariamora** [comment]
No... correct files are ms0020_neg_4703_01.jpg and ms0020_neg_4703_02.jpg.

**Comment by rmjaffe** [comment]
OK, I'm confused.  What I'm seeing in the DAMS is that:

Untitled: Copy series for 1947 C.S.F.A.assignment #1  = ms0020_neg_4703_01.tif
Untitled: Copy series for 1947 C.S.F.A.assignment #2 = ms0020_neg_4703_02.tif

Are these metadata records and files mismatched as well (in addition to the mismatch Sue originally reported, "Pirkle Jones sitting with camera case, San Francisco" which does not match image ms0020_neg_4764.tif)?

**Comment by tmariamora** [comment]
I may need to discuss in person to ensure we're on the same page but to me it looks like what is in the DAMS is indeed correct. I'm looking at https://digitalcollections.library.ucsc.edu/records/6969z139x

and 

https://digitalcollections.library.ucsc.edu/records/37720d36f

Those file names/ titles match those on the original inventory created by Mary. 

ms0020_neg_4764.tif should be an image of Pirkle with his camera case. When I look at the DAMS I see the correct image in the thumbnail (see https://digitalcollections.library.ucsc.edu/catalog?utf8=%E2%9C%93&locale=en&search_field=all_fields&q=ms0020_neg_4764.tif) but when I select that image I see the photo of the drawing (https://digitalcollections.library.ucsc.edu/records/r781wg644) with the metadata for the Pirkle image.

**Comment by rmjaffe** [comment]
That is so weird!  I hadn't noticed that the thumbnail (the image of PJ and the camera case) was correct but the image on the item page is not.  So it's not really the metadata that needs updating, rather we need to look into why the wrong image is being displayed on the item page. @chesleyperry Is this a more accurate description of the error?  Should this ticket be passed on to Ned or pushed back into Small Fixes?

**Comment by chesleyperry** [comment]
That is weird.  I haven't seen this type of issue yet.  I think the title for this should be changed to something like search results thumbnail/item page image mismatch and then sent over to Ned.

**Comment by rmjaffe** [comment]
Updated the title and description of this ticket and pushed it back into small fixes.

**Comment by NedHenry** [comment]
I have confirmed that this is caused by the presence of incorrect cached images on the image server. There are cached derivative images that do not correspond to the fileset that they are supposedly created from. This should never be the case. It is confusing how this happened, but it is almost certainly something I did myself while fixing a previous issue with the image server's file loading algorithm. At one point, the image server was cutting off the last digit of the image id, making it possible for multiple images with similar IDs to overwrite one another's derivatives. I fixed this problem, but apparently some of the overwritten derivative images are still there. This isn't a problem that will ever happen with new ingests, now that we have the ingest workflow and image server working properly.

To fix the problem, I have to delete and recreate the cached derivative images on the image server. It is hard to tell which images might be affected without manually looking through them all, so the best solution is probably to recreate all the image server's jpg derivatives for all the images in the DAMS overnight. I'll plan to do that tonight.
This doesn't take as long as it sounds, because we are just creating jpg derivatives from jpeg2000 files. We don't have to process the TIF files again, which is the slow part.
